testing/unittests/fxpptests.cpp

```diff
@@ -274,141 +210,80 @@ class fxppParserTests : public CppUnit::TestFixture
+          CPPUNIT_ASSERT_EQUAL_MESSAGE("End tag uri mismatch", std::string(uri), std::string(etag.getUri()));
           break;
         }
       }
-      if (!streq(input, output))
-      {
-        fprintf(stdout, "\nTest(%s) input not equal to output:\n    in:  '%s'\n    out: '%s'", cfg.name, input.str(), output.str());
-        failed = true;
-      }
-      return !failed;
+      CPPUNIT_ASSERT_EQUAL_MESSAGE("Input not equal to output", std::string(input), std::string(output));
     }
 
     void testSkipDocument()
     {
-      const char* test = "skip-document";
       std::unique_ptr<IFragmentedXmlPullParser> parser(fxpp::createParser());
       const char* input = R"!!!(<Root><Child/></Root>)!!!";
       parser->setInput(input, int(strlen(input)));
       try
       {
         parser->skipSubTree();
-        if (parser->next() != XmlPullParser::END_DOCUMENT)
-        {
-          fprintf(stdout, "\nTest(%s) did not leave parser at expected position", test);
-          CPPUNIT_ASSERT(false);
-        }
+        CPPUNIT_ASSERT_MESSAGE("Parser not left at expected position", parser->next() == XmlPullParser::END_DOCUMENT);
       }
       catch (XmlPullParserException& xe)
       {
-        fprintf(stdout, "\nTest(%s) failed with exception: %s", test, xe.what());
-        CPPUNIT_ASSERT(false);
+        CPPUNIT_FAIL(VStringBuffer("Parser threw an exception: %s", xe.what()));
       }
     }
 
     void testSkipRoot()
     {
-      const char* test = "skip-root";
       std::unique_ptr<IFragmentedXmlPullParser> parser(fxpp::createParser());
       const char* input = R"!!!(<Root><Child/></Root>)!!!";
       parser->setInput(input, int(strlen(input)));
       try
       {
         parser->next();
         parser->skipSubTree();
-        if (parser->next() != XmlPullParser::END_DOCUMENT)
-        {
-          fprintf(stdout, "\nTest(%s) did not leave parser at expected position\n", test);
-          CPPUNIT_ASSERT(false);
-        }
+        CPPUNIT_ASSERT_MESSAGE("Parser not left at expected position", parser->next() == XmlPullParser::END_DOCUMENT);
       }
       catch (XmlPullParserException& xe)
       {
-        fprintf(stdout, "\nTest(%s) failed with exception: %s\n", test, xe.what());
-        CPPUNIT_ASSERT(false);
+        CPPUNIT_FAIL(VStringBuffer("Parser threw an exception: %s", xe.what()));
       }
     }
 
     void testSkipMalformed()
     {
-      const char* test = "skip-malformed";
       std::unique_ptr<IFragmentedXmlPullParser> parser(fxpp::createParser());
       const char* input = R"!!!(<Root>)!!!";
       parser->setInput(input, int(strlen(input)));
       try
       {
         parser->skipSubTree();
-        fprintf(stdout, "\nTest(%s) completed without expected exception expected position\n", test);
-        CPPUNIT_ASSERT(false);
+        CPPUNIT_FAIL("Parser did not throw expected exception");
```

Comment:
I think there is a macro that would allow you to assert an `XmlPullParserException` is thrown for the line above.